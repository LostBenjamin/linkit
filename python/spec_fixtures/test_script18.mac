
%include symbols

$render-time 100
$width0 15
$width1 90
$width2 75
$time1 50
$time2 200
$color 1
$seq-type sqs
$fade-type ffd
$step 2

[fancy 10]
  app
  (init)

[init]
  <DRAW-PLUS>:drw
  <render-time>,<render>:sch
  (start)

[start]
  (((worm-start 3)))

#(((worm 1 <time1> <time2> <color> <step>)))

((worm 0 50 200 0 1))
((worm 1 45 195 3 2))
((worm 2 40 190 4 3))

[[worm-start X
  (worm-start-X)
]]

[[worm X TIME1 TIME2 COLOR STEP
  [worm-start-X]
    {seq0-X},<width0>:<seq-type>
    {seq1-X},<width1>:<seq-type>
    {seq2-X},<width2>:<seq-type>
    (worm-draw-X TIME1)
    (worm-advance-X TIME2)

  [worm-draw-X]
    <seq1-X>,<MACRO-SEQ>,<worm-math-X>:snw
    pos
    COLOR:pal
    <fade-type>
    flo
    rst

  [worm-math-X]
    <seq0-X>:seq
    psh
    <seq2-X>,<GET-CURRENT>:seq
    psh
    add

  [worm-advance-X]
    <seq2-X>,0,STEP:seq
]]

