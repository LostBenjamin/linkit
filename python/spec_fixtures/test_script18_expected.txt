10:set:app:18:run
11:set:1,-4,19:snw:pos:0:pal:ffd:flo:rst
13:set:4,-4,20:snw:pos:3:pal:ffd:flo:rst
15:set:7,-4,21:snw:pos:4:pal:ffd:flo:rst
17:set:25:run:22:run:28:run
18:set:1:drw:100,78:sch:17:run
19:set:0:seq:psh:2,-1:seq:psh:add
20:set:3:seq:psh:5,-1:seq:psh:add
21:set:6:seq:psh:8,-1:seq:psh:add
22:set:3,15:sqs:4,90:sqs:5,75:sqs:45,13:sch:195,31:sch
25:set:0,15:sqs:1,90:sqs:2,75:sqs:50,11:sch:200,32:sch
28:set:6,15:sqs:7,90:sqs:8,75:sqs:40,15:sch:190,33:sch
31:set:5,0,2:seq
32:set:2,0,1:seq
33:set:8,0,3:seq
78:set:flu

